lib/options_validator.js

```diff
@@ -106,11 +106,10 @@ function validate(
       );
     }
 
-    if (
-      optionDefaults &&
-      optionDefaults.hasOwnProperty(optionName) &&
-      optionDefaults[optionName] != null
-    ) {
+    if (optionDefaults && optionDefaults.hasOwnProperty(optionName)) {
+      if (optionDefaults[optionName] == null) {
+        throw new Error(`A dynamic default of null or undefined was provided for [${optionName}].`);
```

Comment:
I'm a little scared of this. Is it not possible for a dynamic default to be null b/c the inherited value is null?

Reply:
Yeah, I meant for this to be a warning not an error. @daprahamian what is an example case where a default is `null`?

Reply:
The `countDocuments should correctly call the given callback` test in `collection_test` currently warns that `serializeFunctions` is null. Based on the tests, it seems that there are many places where the bson options are not set on `this.s.`.

Reply:
It seems like a lot of options can be:

1. If the user provided them for this call, use them,
2. Else if they set them on the mongoClient, use them
3. Else consider the option non-existent, and use a static default

Reply:
@kvwalker let's make a ticket for the follow-on epic, remove the code check for whether the default is null/undefined, and unblock getting the rest of this PR in now.

Reply:
Okay, made NODE-1748 and removed the code.